internal/k8s/k8s.go

```diff
@@ -1,32 +1,48 @@
 package k8s
 
 import (
-	"context"
-
-	"github.com/simskij/xk6-chaos/internal/k8s/pods"
-	"github.com/simskij/xk6-chaos/pkg/k8s/client"
+	"github.com/dop251/goja"
+	"github.com/grafana/xk6-chaos/internal/k8s/pods"
+	"github.com/grafana/xk6-chaos/pkg/k8s/client"
 	"go.k6.io/k6/js/common"
 	"go.k6.io/k6/js/modules"
 )
 
 func init() {
-	modules.Register("k6/x/chaos/k8s", &K8s{
-		Pods: &pods.Pods{},
-	})
+	modules.Register("k6/x/chaos/k8s", K8sRoot{})
+}
+
+var _ modules.Module = K8sRoot{}
+
+type K8sRoot struct{}
+
+func (k K8sRoot) NewModuleInstance(vu modules.VU) modules.Instance {
+	return &K8s{
+		vu: vu,
+	}
 }
 
 // K8s exports all kubernetes related APIs
 type K8s struct {
+	vu   modules.VU
 	Pods *pods.Pods
 }
 
+func (k *K8s) Exports() modules.Exports {
+	return modules.Exports{
+		Named: map[string]interface{}{
+			"Pods": k.XPods,
+		},
+	}
+}
+
 // XPods serves as a constructor of the Pods js class
-func (*K8s) XPods(ctx *context.Context) (interface{}, error) {
-	rt := common.GetRuntime(*ctx)
+func (k *K8s) XPods(goja.ConstructorCall) goja.Object {
```

Comment:
This should also be a pointer return, i.e. `*goja.Object`.

Reply:
see https://github.com/grafana/xk6-chaos/pull/5#discussion_r872090834